### PR TITLE
♻️ introducing structured logging (part 2)

### PR DESCRIPTION
**Pull request by matusdrobuliak66**
## What do these changes do?
 - Continuation on #4114 
 - Normalizing log format across the whole Python stack (excluding Director-v0 and osparc-gateway-server)
 - Introducing env variable LOG_FORMAT_LOCAL_DEV_ENABLED
   -  Reasoning:
     - new log format double escapes new line character to preserve it as one log message, which is then parsed back in the Graylog to be nicely human readable
     - when LOG_FORMAT_LOCAL_DEV_ENABLED is enabled during local development, log messages are not double escaped + coloring of log_level is introduced for a better experience during local development

## Related issue/s

- https://github.com/ITISFoundation/osparc-simcore/issues/4079
- https://github.com/ITISFoundation/osparc-simcore/pull/4114
- closes https://github.com/ITISFoundation/osparc-simcore/issues/2157

### CODE REVIEW

File: .env-devel [hunk 1]
```diff
@@ -119,3 +119,5 @@ WEBSERVER_STUDIES_ACCESS_ENABLED=0
 # PYTHONTRACEMALLOC=1
 # PYTHONASYNCIODEBUG=1
 # AIODEBUG_SLOW_DURATION_SECS=0.25
+
+LOG_FORMAT_LOCAL_DEV_ENABLED=1
```
Comment (sanderegg) [inline .env-devel:123]: small improvement: put it alphabetically sorted

File: services/web/server/src/simcore_service_webserver/log.py [hunk 2]
```diff
@@ -18,13 +18,18 @@
 )
 
 
-def setup_logging(*, level: str | int, slow_duration: float | None = None):
+def setup_logging(
+    *,
+    level: str | int,
+    slow_duration: float | None = None,
+    log_format_local_dev_enabled: bool = False
```
Comment (sanderegg) [inline services/web/server/src/simcore_service_webserver/log.py:25]: always wondering at these defaults. in this case do we really need them? there is just the one call from cli.py right? why not just remove all the defaults if they are anyway set?

**Review by sanderegg — APPROVED** [review]
thanks. a few comments. This will help a lot

**Review by mrnicegyu11 — APPROVED** [review]
looks good from my side, thanks a lot

File: packages/service-library/src/servicelib/logging_utils.py [hunk 3]
```diff
@@ -52,21 +52,23 @@ class CustomFormatter(logging.Formatter):
     2. Overrides 'filename' with the value of 'file_name_override', if it exists.
     """
 
-    def __init__(self, fmt, color_log_enabled: bool = False):
+    def __init__(self, fmt, log_format_local_dev_enabled):
```
Comment (GitHK) [inline packages/service-library/src/servicelib/logging_utils.py:55]: please add typing

**Review by GitHK — APPROVED** [review]

File: services/agent/src/simcore_service_agent/core/application.py [hunk 4]
```diff
@@ -26,7 +26,7 @@ def _setup_logger(settings: ApplicationSettings):
     # SEE https://github.com/ITISFoundation/osparc-simcore/issues/3148
     logging.basicConfig(level=settings.LOGLEVEL.value)  # NOSONAR
     logging.root.setLevel(settings.LOGLEVEL.value)
-    config_all_loggers()
+    config_all_loggers(settings.AGENT_VOLUMES_LOG_FORMAT_LOCAL_DEV_ENABLED)
```
Comment (pcrespov) [inline services/agent/src/simcore_service_agent/core/application.py:29]: NICE!

**Review by pcrespov — APPROVED** [review]
thx!
please double check with ops the issue I raise about ``.env-devel``

File: .env-devel [hunk 5]
```diff
@@ -48,6 +48,7 @@ INVITATIONS_PORT=8000
 INVITATIONS_SECRET_KEY='REPLACE_ME_with_result__Fernet_generate_key='
 INVITATIONS_USERNAME=admin
 
+LOG_FORMAT_LOCAL_DEV_ENABLED=1
```
[the remark this answers is not in the document]
Comment (matusdrobuliak66) [reply .env-devel:51]: I check with ops, it should be fine as there is predefined default to be 0.

**Pull request merged**